apiserver/facades/agent/uniter/networkinfoiaas.go

```diff
@@ -274,44 +281,50 @@ func (n *NetworkInfoIAAS) populateMachineNetworkInfos() error {
 	// the IP.
 	// TODO (manadart 2020-02-21): As mentioned above, this is not required
 	// when we have subnets populated for all providers.
-	if r, ok := results[network.AlphaSpaceId]; !ok && spaceSet.Contains(network.AlphaSpaceId) {
+	if _, ok := n.machineNetworkInfos[network.AlphaSpaceId]; !ok && spaceSet.Contains(network.AlphaSpaceId) {
 		if privateLinkLayerAddress != nil {
-			r.Info, _ = addAddressToResult(r.Info, privateLinkLayerAddress)
+			n.addAddressToResult(network.AlphaSpaceId, privateLinkLayerAddress)
```

Comment:
Minor nit; you could continue here and de-indent the following else block.

Reply:
This is not actually inside the for loop, so we can't continue.